library/alloc/src/rc.rs

```diff
@@ -1915,7 +1915,7 @@ impl From<&str> for Rc<str> {
 
 #[cfg(not(no_global_oom_handling))]
 #[stable(feature = "shared_from_slice", since = "1.21.0")]
-impl From<String> for Rc<str> {
+impl<A: Allocator> From<String<A>> for Rc<str> {
```

Comment:
I think this conversion has to produce a new allocation anyway, because of the `RcBox` header alignment.

As it is currently implemented, `From<String> for Rc<str>` just derefs the String and uses `From<&str> for Rc<str>`, and this PR does not currently change that.

`Rc`/`Arc` don't currently support allocators however, so if `From<String<A>> for Rc<str, A>` is wanted, I could remove these impls from this PR and leave them to future PRs.

The corresponding impls don't currently exist for `Vec` (i.e. there is only `From<Vec<T, Global>> for Rc<[T]>`, (which also creates a new allocation)), so it may be better to remove these impls from this PR anyway. I'm not sure though.

Reply:
I have removed the alloc-ification of these impls from this PR to leave room for making Rc/Arc allocator-aware in the future .